misc/atrium-specs/src/commonMain/kotlin/ch/tutteli/atrium/specs/integration/AbstractIterableNotToContainValuesExpectationsTest.kt

```diff
@@ -0,0 +1,156 @@
+import ch.tutteli.atrium.specs.integration.IterableToContainEntriesSpecBase.Companion.mismatchedIndex
+import ch.tutteli.atrium.specs.integration.IterableToContainSpecBase.Companion.anElementWhichEquals
+import ch.tutteli.atrium.specs.integration.IterableToContainSpecBase.Companion.fluentEmpty
+import ch.tutteli.atrium.specs.integration.IterableToContainSpecBase.Companion.mismatches
+import ch.tutteli.atrium.specs.integration.IterableToContainSpecBase.Companion.notToContainDescr
+import ch.tutteli.atrium.specs.integration.IterableToContainSpecBase.Companion.oneToSeven
+import ch.tutteli.atrium.specs.integration.IterableToContainSpecBase.Companion.oneToSevenNullable
+import ch.tutteli.atrium.specs.integration.IterableToContainSpecBase.Companion.separator
+import ch.tutteli.atrium.specs.integration.utils.SubjectLessTestData
+import ch.tutteli.atrium.testfactories.TestFactory
+import ch.tutteli.atrium.translations.DescriptionIterableLikeExpectation
+
+@Suppress("FunctionName")
+abstract class AbstractIterableNotToContainValuesExpectationsTest(
+    private val notToContainValuesSpec: Fun2<Iterable<Double>, Double, Array<out Double>>,
+    private val notToContainNullableValuesSpec: Fun2<Iterable<Double?>, Double?, Array<out Double?>>,
+    private val notToHaveElementsOrNoneFunNameSpec: String,
+) : ExpectationFunctionBaseTest() {
+
+    @TestFactory
+    fun subjectLessTest() = subjectLessTestFactory(
+        SubjectLessTestData(
+            notToContainValuesSpec.forSubjectLessTest(2.3, arrayOf())
+        ),
+        SubjectLessTestData(
+            notToContainNullableValuesSpec.forSubjectLessTest(2.3, arrayOf())
+        )
+    )
+
+    fun Expect<Iterable<Double?>>.notToContainNullableFun(a: Double?, vararg aX: Double?) =
+        notToContainNullableValuesSpec(this, a, aX)
+
+    val anElementWhichIsWithIndent = "$indentRootBulletPoint$listBulletPoint$anElementWhichEquals"
+
+
+    @TestFactory
+    fun empty_collection__4__throws() = nonNullableCases(
+        notToContainValuesSpec,
+        notToContainNullableValuesSpec
+    ) { notToContainValuesFun ->
+        expect {
+            expect(fluentEmpty()).notToContainValuesFun(4.0, emptyArray())
+        }.toThrow<AssertionError> {
+            message {
+                toContainRegex(
+                    "$hasANextElement$separator" +
+                        "$indentRootBulletPoint\\Q$explanatoryBulletPoint\\E$notToContainDescr: $separator" +
+                        "$indentListBulletPoint$anElementWhichIsWithIndent: 4.0.*",
+                    "$hintBulletPoint${
+                        DescriptionIterableLikeExpectation.USE_NOT_TO_HAVE_ELEMENTS_OR_NONE.getDefault()
+                            .format(notToHaveElementsOrNoneFunNameSpec)
+                    }"
+                )
+            }
+        }
+    }
+
+    fun happy__cases__do__not__throw() = testFactory(notToContainValuesSpec) { notToContainFun ->
```

Comment:
please execute test locally and check if all are executed. 
Also please review the naming of the methods:
```suggestion
    fun happy_cases__do_not_throw() = testFactory(notToContainValuesSpec) { notToContainFun ->
```